updates/13.sql

```diff
@@ -0,0 +1,7 @@
+--avoid foreign key check for this session, otherwise the alter table statement will not work
+SET FOREIGN_KEY_CHECKS = 0;
```

Comment:
Disabling foreign key checks is a bad thing. The foreign keys are guaranteeing the consistency of the database.

If you add auto_increment on an existing column, the IDs will be reindexed. So if there are deleted values, the ids will no longer match the foreign keys.

For example, we have 11 Dichtung and 13 Testament, so 12 is a gap which has been deleted. If you add auto_increment now, 13 will be reindexed as 12 which means all foreign references will get broken.

So there must be a better way to achieve this. I found a stackoverflow article on this:
https://stackoverflow.com/questions/740358/reorder-reset-auto-increment-primary-key

Maybe the AUTO_INCREMENT value needs to be set explicitly to the max current value of the database?

Reply:
I have changed it as suggested in the post of stackoverflow.
But there is still an error message:
`Error Code: 1451. Cannot delete or update a parent row: a foreign key constraint fails (`neg`.`einzelbeleg`, CONSTRAINT `einzelbeleg_QuelleGattungID` FOREIGN KEY (`QuelleGattungID`) REFERENCES `selektion_quellengattung` (`ID`))
`
The following statement return 0 rows. So the column QuelleGattungID of einzelbelg seems to be fine:
`SELECT ID FROM einzelbeleg WHERE QuelleGattungID NOT IN (SELECT `ID` FROM `selektion_quellengattung`);`

Not sure how to solve it

Reply:
This SELECT will only prove that all IDs still exist in the other table, but not that the actual values are the same as before the operation. But you're right, this is indeed a hard nut to crack. I will have another look, but it might take some time.

Reply:
After some more research i found this article:
https://stackoverflow.com/questions/50117614/auto-increment-primary-key-and-update-its-references-in-other-tables-at-the-same

So maybe we need to delete the existing foreign key & recreate it using ON UPDATE CASCADE before we add the AUTO INCREMENT to the corresponding column?

Reply:
Yes. It works now